src/libs/actions/Report.ts

```diff
@@ -2713,6 +2713,7 @@ function openReportFromDeepLink(url: string) {
                             return;
                         }
 
+                        Onyx.disconnect(connectionID);
```

Comment:
@bernhardoj I think we should disconnect after the navigate function. Is there any problem with that?

Reply:
It shouldn't matter whether we call it before or after. Disconnecting it won't cancel the navigation. In fact, I was thinking whether we should disconnect it earlier or not. For example, if the user is anonymous and trying to access deep link, we return early and redirect them to the sign-in modal and the connection is still there, so when the onboarding onyx data changes, the deep link logic is triggered again. I think this applies to every early return inside the logic, but I haven't tested it yet though.

https://github.com/Expensify/App/blob/156f0a2bd0e94558ad621caa701bd749dd124bfe/src/libs/actions/Report.ts#L2677-L2684

```
if (onboarding) {
    Onyx.disconnect(...);
}
```

Reply:
Yeah, let's move this to the top of the callback please and add a comment explaining why it's important to disconnect.

Reply:
Updated